pkg/scheduler/internal/cache/cache.go

```diff
@@ -220,6 +223,7 @@ func (cache *cacheImpl) UpdateSnapshot(nodeSnapshot *Snapshot) error {
 			// all the nodes are updated before the existing snapshot. We are done.
 			break
 		}
+		updateUsedPVCSet = true
```

Comment:
Can we make it conditional, with a minimum calculation cost like `updateNodesHavePodsWithAffinity`?

Reply:
Sure, added an additional check
```go
if !reflect.DeepEqual(existing.PVCRefCounts, clone.PVCRefCounts) {
	updateUsedPVCSet = true
}
```
in `UpdateSnapshot()`

and in `types.go` when we clone `NodeInfo`, updated to do a proper clone of the `PVCRefCounts` map rather than updating the reference so that we can perform this comparison when taking snapshot.

Reply:
Sure, added an additional check
```go
if !reflect.DeepEqual(existing.PVCRefCounts, clone.PVCRefCounts) {
	updateUsedPVCSet = true
}
```
in `UpdateSnapshot()`

and in `types.go` when we clone `NodeInfo`, updated to do a proper clone of the `PVCRefCounts` map rather than updating the reference so that we can perform this comparison when taking snapshot.